maas/data_source_maas_vm_host.go

```diff
@@ -0,0 +1,159 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/maas/gomaasclient/client"
+)
+
+func dataSourceMaasVMHost() *schema.Resource {
+	return &schema.Resource{
+		Description: "Provides details about an existing MAAS VM host.",
+		ReadContext: dataSourceVMHostRead,
+
+		Schema: map[string]*schema.Schema{
+			"certificate": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Certificate to use for power control of the LXD VM host.",
+			},
+			"cpu_over_commit_ratio": {
+				Type:        schema.TypeFloat,
+				Computed:    true,
+				Description: "The VM host CPU overcommit ratio.",
+			},
+			"default_macvlan_mode": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The VM host default macvlan mode. Supported values are: `bridge`, `passthru`, `private`, `vepa`.",
+			},
+			"key": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+				Description: "Key to use for power control of the LXD VM host.",
+			},
+			"memory_over_commit_ratio": {
+				Type:        schema.TypeFloat,
+				Computed:    true,
+				Description: "The VM host RAM memory overcommit ratio.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The VM host name.",
+			},
+			"pool": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The VM host pool name.",
+			},
+			"power_address": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Address that gives MAAS access to the VM host power control.",
+			},
+			"power_pass": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+				Description: "User password to use for power control of the VM host.",
+			},
+			"power_user": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "User name to use for power control of the VM host.",
+			},
+			"resources_cores_total": {
```

Comment:
Let's also add `project` since this is another parameter for LXD VM hosts. The default one is `maas` and it means that MAAS will compose LXD VMs inside that project. Afterwards, please also update the docs to reflect the additional field.

Reply:
Are you sure the default one is `maas` @skatsaounis ?
 
Looking at documentation it seems that default is "default"

```text
 :param project: Optional. For ``lxd`` VM hosts, the
project that MAAS will manage. If not provided, the ``default`` project
will be used. If a nonexistent name is given, a new project with that
name will be created.
:type project: String
```

https://maas.io/docs/how-to-use-lxd-projects

Reply:
correct observation. The default is `default`. It's not important for the implementation of the data source but better be accurate. Thank you for :slightly_smiling_face:

Reply:
Hey @skatsaounis I will do when I can as I stop this project for a moment as you may have seen. I'm not so comfortable to write and run needed tests. I just push some code recently because I was reinstalling my gear and wanted to avoid loosing stuff (but didn't think it will reflect directly on this PR).
So work is not finished yet